sdk/helper/consts/consts.go

```diff
@@ -34,4 +34,7 @@ const (
 	ReplicationResolverALPN = "replication_resolver_v1"
 
 	VaultEnableFilePermissionsCheckEnv = "VAULT_ENABLE_FILE_PERMISSIONS_CHECK"
+
+	// VaultOverrideVersionEnv used in testing to override the reported running version
+	VaultOverrideVersionEnv = "VAULT_TESTING_OVERRIDE_VERSION"
```

Comment:
I went down this path initially, but didn't like that I had to compile a few versions of the plugin in order to get different versions, since it makes the tests a little slower. But, I think you're right, that it would be better to have them be immutable artifacts. I'll rework this to use a fixed variable.